src/webgpu/api/operation/resource_init/check_texture/by_ds_test.ts

```diff
@@ -0,0 +1,197 @@
+import { assert } from '../../../../../common/framework/util/util.js';
+import { GPUTest } from '../../../../gpu_test.js';
+import { CheckContents } from '../texture_zero_init.spec.js';
+
+function fullscreenVertexModule(device: GPUDevice) {
+  return device.createShaderModule({
+    code: `
+    [[builtin(position)]] var<out> Position : vec4<f32>;
+    [[builtin(vertex_idx)]] var<in> VertexIndex : i32;
+
+    [[stage(vertex)]]
+    fn main() -> void {
+      const pos : array<vec2<f32>, 3> = array<vec2<f32>, 3>(
+        vec2<f32>(-1.0, -3.0),
+        vec2<f32>( 3.0,  1.0),
+        vec2<f32>(-1.0,  1.0));
+      Position = vec4<f32>(pos[VertexIndex], 0.0, 1.0);
+      return;
+    }
+    `,
+  });
+}
+
+function getDepthTestReadbackPipeline(
```

Comment:
nit: I don't think this would be readback because it doesn't sample the texture. `getDepthTestEqualPipeline`?